dev/Interop/StoragePickers/PickerCommon.cpp

```diff
@@ -0,0 +1,213 @@
+        if (!knownFolder)
+        {
+            knownFolderManager->GetFolder(FOLDERID_Documents, knownFolder.put());
+        }
+        if (!knownFolder)
+        {
+            return nullptr;
+        }
+
+        winrt::com_ptr<IShellItem> defaultFolder{};
+        hr = knownFolder->GetShellItem(0, IID_PPV_ARGS(defaultFolder.put()));
+        if (FAILED(hr))
+        {
+            return nullptr;
+        }
+
+        return defaultFolder;
+    }
+
+
+    winrt::hstring FormatExtensionWithWildcard(winrt::hstring extension)
+    {
+        if (!extension.empty() && extension[0] == L'*')
+        {
+            return extension;
+        }
+        else
+        {
+            return L"*" + extension;
+        }
+    }
+
+    winrt::hstring JoinExtensions(winrt::Windows::Foundation::Collections::IVectorView<winrt::hstring> extensions)
+    {
+        winrt::hstring result;
+        bool first = true;
+        for (const auto& ext : extensions)
+        {
+            if (first)
+            {
+                result = FormatExtensionWithWildcard(ext);
+                first = false;
+            }
+            else
+            {
+                result = result + L";" + FormatExtensionWithWildcard(ext);
+            }
+        }
+        return result;
+    }
+}
+
+
+namespace PickerCommon {
+
+    using namespace winrt;
+
+    bool IsHStringNullOrEmpty(winrt::hstring value)
+    {
+        return value.empty();
+    }
+
+    winrt::hstring GetPathFromShellItem(winrt::com_ptr<IShellItem> shellItem)
+    {
+        wil::unique_cotaskmem_string filePath;
+        check_hresult(shellItem->GetDisplayName(SIGDN_FILESYSPATH, filePath.put()));
+        return winrt::hstring{ filePath.get() };
+    }
+
+    std::vector<COMDLG_FILTERSPEC> CaptureFilterSpec(std::vector<winrt::hstring>& buffer, winrt::Windows::Foundation::Collections::IVectorView<winrt::hstring> filters)
+    {
+        std::vector<COMDLG_FILTERSPEC> result(filters.Size());
+        buffer.clear();
+        buffer.reserve(filters.Size() * (size_t)2);
+        for (auto filter : filters)
+        {
+            auto ext = FormatExtensionWithWildcard(filter);
+            buffer.push_back(filter);
+            buffer.push_back(ext);
+        }
+        for (size_t i = 0; i < filters.Size(); i++)
+        {
+            result.at(i) = { buffer.at(i * 2).c_str(), buffer.at(i * 2 + 1).c_str() };
+        }
+
+        if (result.size() == 0)
+        {
+            result.push_back({ L"All Files", L"*.*" });
+        }
+        return result;
+    }
+
+    std::vector<COMDLG_FILTERSPEC> CaptureFilterSpec(std::vector<winrt::hstring>& buffer, winrt::Windows::Foundation::Collections::IMapView<winrt::hstring, winrt::Windows::Foundation::Collections::IVector<hstring>> filters)
+    {
+        std::vector<COMDLG_FILTERSPEC> result(filters.Size());
+        buffer.clear();
+        buffer.reserve(filters.Size() * 2ull);
```

Comment:
>2ull [](http://example.com/codeflow?start=40&length=4)

nit: curious that this is different than the other CaptureFilterSpec

Reply:
thank you for your suggestions, fixed to use same `(size_t)` conversion